### PR TITLE
[BugFix] Fix discrete SAC log-prob

### PR DESCRIPTION
**Pull request by vmoens**
cc @matteobettini

### CODE REVIEW

**Comment by matteobettini** [comment]
I don't get the fix.

We need both prob and log_prob

We also need to rerun the example training run for discrete SAC and confirm the returns curve match

File: torchrl/objectives/sac.py [hunk 1]
```diff
@@ -1201,8 +1201,7 @@ def _actor_loss(
         # get probs and log probs for actions
         with self.actor_network_params.to_module(self.actor_network):
             dist = self.actor_network.get_dist(tensordict.clone(False))
-        prob = dist.probs
-        log_prob = prob.clamp_min(torch.finfo(prob.dtype).resolution)
+        log_prob = dist.logits
```
Comment (matteobettini) [inline torchrl/objectives/sac.py:1204]:
```suggestion
        prob = dist.probs
        log_prob = dist.logits
```

**Review by matteobettini — APPROVED** [review]
LGTM

If we want to be extra careful we should run the discrete_sac example to check

**Comment by matteobettini** [comment]
This fix actually changed the behavior and is breaking the CI of BenchMARL for discrete SAC with MaskedCategorical.

What happens is that if `dist.probs` returns  `[[1.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000,
          0.0000, 0.0000, 0.0000]`

`dist.logits` will give `[ 0.0000,    -inf,    -inf,    -inf,    -inf,    -inf,    -inf,
             -inf,    -inf,    -inf,    -inf]`

while `torch.log(torch.where(dist.probs == 0, 1e-8, dist.probs))` gives ` [[  0.0000, -18.4207, -18.4207, -18.4207, -18.4207, -18.4207, -18.4207,
          -18.4207, -18.4207, -18.4207, -18.4207]`

Can we return to the previous behavior?

**Comment by vmoens** [comment]
Not sure returning to the previous behaviour is appropriate. Why is this a problem?
-18 log prob is roughly equivalent to -inf and is more numerically stable IMO

**Comment by matteobettini** [comment]
The -inf causes a but in MaskedCategorical https://github.com/facebookresearch/BenchMARL/actions/runs/7278697272/job/19833445677

If you prefer to adapt how the logits are computed in `MaskedCategorical`, that's should be a viable solution too

**Comment by matteobettini** [comment]
I wonder what was the problem of the previous behavior tho.
We were just clamping the probs and computing the log, without calling `dist.logits`

**Comment by vmoens** [comment]
Efficiency: we were recomputing something already computed by the dist
Stability: the log-probs should be normalised. The dist does that, we didn't.

And the code previously was seriously ugly
